### DISCUSSION TITLE
Docker Container for easy use

### DISCUSSION THREAD

**Discussion by edward8e** [anidl/multi-downloader-nx#278] [Ideas]
I thought it would be nice to have the multi-downloader-nx inside of a Docker container since this might make it easy run without needed to deal with any configuration or setup. This is still a work in progress and have a first working draft.

I created a [fork](https://github.com/edward8e/docker-multi-downloader-nxl) which you may checkout. 

I uploaded the built Docker Container on DockerHub if anyone wishes to test it out.
[https://hub.docker.com/r/edward8e/multi-downloader-nx](https://hub.docker.com/r/edward8e/multi-downloader-nx)

## Instructions: 
```docker run -v $(PWD)/videos:/app/videos edward8e/multi-downloader-nx:latest```

## Dockerfile
```
FROM node  AS builder 
WORKDIR "/app"
COPY . .

# Install 7z for packaging

RUN apt-get update
RUN apt-get install p7zip-full -y

# Update bin-path for docker/linux

RUN echo 'ffmpeg: "./bin/ffmpeg/ffmpeg"\nmkvmerge: "./bin/mkvtoolnix/mkvmerge"' > /app/config/bin-path.yml

#Build AniDL

RUN npm i
RUN npm run tsc false
RUN npm run build-ubuntu-cli

# Move build to new Clean Image

FROM node
WORKDIR "/app"
COPY --from=builder /app/lib/_builds/multi-downloader-nx-ubuntu64-cli ./

# Install mkvmerge and ffmpeg

RUN mkdir -p /app/bin/mkvtoolnix
RUN mkdir -p /app/bin/ffmpeg

RUN apt-get update
RUN apt-get install mkvtoolnix -y
RUN apt-get install ffmpeg -y

RUN mv /usr/bin/mkvmerge /app/bin/mkvtoolnix/mkvmerge
RUN mv /usr/bin/ffmpeg /app/bin/ffmpeg/ffmpeg

ENTRYPOINT ["tail", "-f", "/dev/null"]
```
Any feedback or suggestions would be great! Also might be nice to eventually move this to the main project.

**Comment by izu-co** [comment]
Fell free to open a PR - but I am fairly new to docker so I can’t really help.
This would require docker and I think the precompiled versions are easier to use

**Reply by edward8e** [reply]
Sounds good, will open one up!
